AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasNodeInspector.py

```diff
@@ -0,0 +1,109 @@
+from consts.general import (WAIT_TIME_SEC_1, )
+from enum import IntEnum
+
+class BooleanCheckBoxValues(IntEnum): # get a better name for this
+    Off = 0
+    On = 1
+
+class QtPyScriptCanvasNodeInspector:
+
+    def __init__(self, sc_editor):
+        self.node_inspector = sc_editor.sc_editor.findChild(QtWidgets.QDockWidget, NODE_INSPECTOR_QT)
+
+    def change_variable_initial_value(self, variable_name: str, variable_type: str,  new_value) -> None:
+        """
+        Function for changing the initial value of a variable. This function will have branching behavior since variables
+        can have many different types.
+
+        Note: This function assumes the desired node is already in focus and visible to the node inspector
+
+        IMPORTANT NOTE: Only the basic variable types are supported by this function. EntityId not supported at the moment
+            Basic types: "Boolean", "Color", "Number", "String", "Transform", "Vector2", "Vector3", "Vector4"
+
+        Basic type new_value structures
+            Boolean:
+            Color, String: String
+            Number: float
+            Transform: list[math.Vector3, math.Vector3, float]
+            Vectors use corresponding math library vector
+
+        params variable_name: the name of the variable that's selected
+        params variable_type: the type of the variable being configured.
+        params new_value: the new value you want to initialize the variable with
+
+        returns None
+        """
+
+        variable_qobject = self.node_inspector.findChild(QtWidgets.QFrame, f"{variable_name}")
+
+        assert variable_qobject is not None, f"Variable by name {variable_name} was not found."
+
+        match variable_type:
+            case "Boolean":
+                assert type(new_value) == BooleanCheckBoxValues, f"New value: {new_value} is invalid for this variable type:{variable_type}"
+                self.__change_variable_value_Boolean(variable_qobject, new_value)
+            case "Color", "String":
+                assert type(new_value) == str, f"New value: {new_value} is invalid for this variable type:{variable_type}"
+                self.__change_variable_value_string(variable_qobject, new_value)
+            case default:
+                assert False, "An invalid variable type was provided"
+
+    def __change_variable_value_Boolean(self, variable_qobject: QtWidgets.QFrame, new_value: BooleanCheckBoxValues) -> None:
+        """
+        helper function for the change_variable initial_value branch (boolean variable type)
+        """
+        check_box = variable_qobject.findChild(QtWidgets.QCheckBox, "")
+        new_value_to_bool = bool(new_value)
+
+        if new_value_to_bool != check_box.isChecked():
+            check_box.click()
+
+
+    def __change_variable_value_string(self, variable_qobject: QtWidgets.QFrame, new_value: str) -> None:
+        """
+        helper function for the change_variable_initial_value branch(color, string variable type)
```

Comment:
Added detail to teh comments. This is for initializing the variable but for types that use a string for initialization. You would think it would only be string types but color type variables use a string for the RGBA values. maybe this is because it also accepts a hexadecimal code? I haven't tested that